### DISCUSSION TITLE
How to wait for cy.request with intercepting response with cy.intercept method?

### DISCUSSION THREAD

**Discussion by 403-html** [cypress-io/cypress#15338] [Questions and Help] [answered]
Hello, I'm asking for your help with waiting.
I do request, and I want to wait for it with intercepting it with my own data but a request is made so it'll always fail. How to solve it?
```
describe("Intercept usage", () => {
  it("Change GET output", () => {
    cy.intercept("GET", "**/users", { statusCode: 200, body: "Worked" }).as(
      "getUsers"
    );
    cy.request({
      method: "GET",
      url: "/users",
    }).then((resp) => console.log(resp));
    cy.wait("@getUsers");
  });
});
```
I expect to get a response.body as "Worked" but request sent, for example, give me JSON of the user.

The same thing doesn't work for the obsolete route method.
```
describe("Route usage", () => {
  beforeEach(() => cy.server());
  afterEach(() => cy.server({ enable: false }));

  it("Change GET output", () => {
    cy.route("**/users", {message: "Worked"}).as("getUsers");
    cy.request({
      method: "GET",
      url: "/users",
    }).then((resp) => console.log(resp));
    cy.wait("@getUsers");
  });
});
```

**Comment by bahmutov** [comment, marked as answer]
You cannot: `cy.request` was specifically made to bypass Cypress interception, https://docs.cypress.io/api/commands/request.html#and

**Reply by 403-html** [reply]
ok, huge thanks!
I got into this before but I forgot to put here answer/close it. Anyway, I made a simple web UI to call requests and intercept them so the problem's solved.
